File: c/collatz-conjecture/collatz_conjecture.c
#include "collatz_conjecture.h"

int steps(int n) {
  if (n < 1)
    return ERROR_VALUE;

  int result = 0;

  while (n != 1) {
    if (n % 2 == 0) {
      n /= 2;
    } else {
      n = 3 * n + 1;
    }

    result++;
  }

  return result;
}

File: python/collatz-conjecture/collatz_conjecture.py
def steps(number: int):
    if number < 1:
        raise ValueError("Only positive integers are allowed")

    result = 0

    while number != 1:
        if number % 2 == 0: number //= 2
        else: number = number * 3 + 1
        result += 1

    return result
